### PR TITLE
Add handler to installed headers

### PR DESCRIPTION
**Pull request by sandro-elsweijer**
**_Describe your changes here:_**
I tried to link against an installed t8code, but it does not work if the handler is not part of the interface

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [x] The reviewer executed the new code features at least once and checked the results manually

- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] New source/header files are properly added to the Makefiles
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [x] The code is covered in an existing or new test case using Google Test

#### Github action

- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [x] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [x] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [x] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [x] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -57,6 +57,7 @@ libt8_installed_headers_forest = \
   src/t8_forest/t8_forest_iterate.h src/t8_forest/t8_forest_partition.h
 libt8_installed_headers_geometry = \
   src/t8_geometry/t8_geometry.h \
+  src/t8_geometry/t8_geometry_handler.hxx \
```
Comment (Davknapp) [inline src/Makefile.am:60]: Wen can remove the t8_geometry_handler from the internal headers then, right?

Comment (sandro-elsweijer) [reply src/Makefile.am:60]: But it is already removed?

File: src/Makefile.am [hunk 2]
```diff
@@ -92,7 +93,6 @@ libt8_installed_headers_default_prism =
 libt8_installed_headers_default_pyramid =
 libt8_internal_headers = \
   src/t8_cmesh/t8_cmesh_trees.h src/t8_cmesh/t8_cmesh_partition.h \
-  src/t8_geometry/t8_geometry_handler.hxx \
```
Comment (sandro-elsweijer) [inline src/Makefile.am:95]: right here

Comment (Davknapp) [reply src/Makefile.am:95]: Sorry, missed that.

**Review by Davknapp — APPROVED** [review]

**Pull request merged**